src/lib/converter/plugins/DynamicModulePlugin.ts

```diff
@@ -72,8 +78,28 @@ export class DynamicModulePlugin extends ConverterComponent {
     private onBeginResolve(context: Context) {
         this.reflections.forEach((reflection) => {
             let name = reflection.name.replace(/"/g, '');
-            name = name.substr(0, name.length - Path.extname(name).length);
-            reflection.name = '"' + this.basePath.trim(name) + '"';
+
+            const currentExtension = Path.extname(name);
+            const hiddenExtensions = (this.hiddenExtension.includes(currentExtension)) ?
+                this.hiddenExtension : [currentExtension, ...this.hiddenExtension];
+
+            hiddenExtensions.forEach((extension) => {
+                if (name.endsWith(extension)) {
+                    name = name.substr(0, name.length - extension.length);
+                }
+            });
+
+            this.hiddenPathPart.forEach((pathPart) => {
+                if (name.endsWith(pathPart)) {
+                    name = name.substr(0, name.length - pathPart.length);
+                }
+            });
+
+            reflection.name = this.basePath.trim(name);
+
+            // in case of mono-repo (node_modules outside the project), trunk the module name after the node_modules folder
+            const nodeModulesRegexp = new RegExp('^(.*)node_modules/');
```

Comment:
Better style to use the literal regex syntax:
```suggestion
            const nodeModulesRegexp = /^(.*)node_modules\//;
```

Reply:
I'm kind of worried about this though, it seems like it could lead to paths appearing to be contained within the project that are external.

Reply:
Ok will fix